lib/modules/platform/azure/index.ts

```diff
@@ -144,23 +148,34 @@ export async function getRawFile(
       return null;
     }
 
-    const versionDescriptor: GitVersionDescriptor = {
-      version: branchOrTag,
-    } satisfies GitVersionDescriptor;
-
-    const item = await azureApiGit.getItem(
-      repoId, // repositoryId
-      fileName, // path
-      undefined, // project
-      undefined, // scopePath
-      undefined, // recursionLevel
-      undefined, // includeContentMetadata
-      undefined, // latestProcessedChange
-      undefined, // download
-      branchOrTag ? versionDescriptor : undefined, // versionDescriptor
-      true, // includeContent
-    );
-
+    // Try to get file from repo with branch, if not found, then try with tag #36835
+    for (const versionType of [GitVersionType.Branch, GitVersionType.Tag]) {
+      const versionDescriptor: GitVersionDescriptor = {
+        version: branchOrTag,
+        versionType,
+      };
+
+      item = await azureApiGit.getItem(
+        repoId, // repositoryId
+        fileName, // path
+        undefined, // project
+        undefined, // scopePath
+        undefined, // recursionLevel
+        undefined, // includeContentMetadata
+        undefined, // latestProcessedChange
+        undefined, // download
+        branchOrTag ? versionDescriptor : undefined, // versionDescriptor
+        true, // includeContent
+      );
+      if (item) {
+        break; // exit loop if item is found
+      } else {
+        logger.debug(
+          `File: ${fileName} not found in ${repoName} with ${versionType}: ${branchOrTag} - trying next versionType`,
```

Comment:
```suggestion
          `File: ${fileName} not found in ${repoName} with ${versionType}: ${branchOrTag}`
```
On the second iteration, we don't need to say we are looking again. Better to remove